test/registry/DefaultRegistryClient.test.ts

```diff
@@ -0,0 +1,56 @@
+import { loggerSpy } from '../testhelpers';
+import { DefaultRegistryClient } from '../../src/registry/DefaultRegistryClient';
+import { NPMRegistryClient } from '../../src/registry/NPMRegistryClient';
+import { FHIRRegistryClient } from '../../src/registry/FHIRRegistryClient';
+
+describe('DefaultRegistryClient', () => {
+  describe('#constructor', () => {
+    beforeEach(() => {
+      loggerSpy.reset();
+      delete process.env.FPL_REGISTRY;
+    });
+
+    it('should make a client with custom registry when it has been specified', () => {
+      process.env.FPL_REGISTRY = 'https://custom-registry.example.org';
+      const defaultClient = new DefaultRegistryClient({ log: loggerSpy.log });
+      expect(defaultClient.clients).toHaveLength(1);
+      expect(defaultClient.clients[0]).toHaveProperty(
+        'endpoint',
+        'https://custom-registry.example.org'
+      );
+      expect(defaultClient.clients[0]).toBeInstanceOf(NPMRegistryClient);
+      expect(loggerSpy.getLastMessage('info')).toBe(
+        'Using custom registry specified by FPL_REGISTRY environment variable: https://custom-registry.example.org'
+      );
+    });
+
+    it('should make a client with fhir registries if no custom registry specified', () => {
+      const defaultClient = new DefaultRegistryClient({ log: loggerSpy.log });
+      expect(defaultClient.clients).toHaveLength(2);
+      expect(defaultClient.clients[0]).toHaveProperty('endpoint', 'https://packages.fhir.org');
+      expect(defaultClient.clients[0]).toBeInstanceOf(FHIRRegistryClient);
+      expect(defaultClient.clients[1]).toHaveProperty(
+        'endpoint',
+        'https://packages2.fhir.org/packages'
+      );
+      expect(defaultClient.clients[1]).toBeInstanceOf(FHIRRegistryClient);
+    });
+
+    it('should make a client by using the first custom client specified if multiple defined', () => {
+      process.env.FPL_REGISTRY = 'https://custom-registry.example.org';
+      const defaultClient = new DefaultRegistryClient({ log: loggerSpy.log });
+      expect(defaultClient.clients).toHaveLength(1);
+      expect(defaultClient.clients[0]).toHaveProperty(
+        'endpoint',
+        'https://custom-registry.example.org'
+      );
+      expect(defaultClient.clients[0]).toBeInstanceOf(NPMRegistryClient);
+      // expect(loggerSpy.getLastMessage('info')).toBe(
+      //   'Using custom registry specified by FPL_REGISTRY environment variable: https://custom-registry.example.org'
+      // );
+      process.env.FPL_REGISTRY = 'https://custom-second-registry.example.org';
+
+      // when hasLoggedCustomRegistry = true ?
+    });
```

Comment:
Line 29 in DefaultRegistryClient keeps track of if it has already logged a custom registry. I removed this test for now, but was attempting to test that part.

Reply:
Oh, I see. The purpose of that is to reduce duplicate logs with the same information. In the old implementation, we called `getCustomRegistry` multiple times -- so we didn't want it to log that message every time. In the new implementation, though, that would only happen if we created multiple `DefaultRegistryClient`s (since we only call `getCustomRegistry` in the constructor) -- so it's a bit less of a problem now.

Still... you could create a test where `process.env.FPL_REGISTRY` is set, then just construct a `DefaultRegistryClient` twice (no need to change the `FPL_REGISTRY`). Then get all the info logs from the `loggerSpy` and confirm only one message contains the custom registry URL.

Reply:
So... as noted in a different comment thread: I was wrong. Because the variable that keeps the state of logging is at the module level, it only logs once during the _entire_ test run. We agreed to just remove the test.